internal/pkg/kafkarest/kafkarest.go

```diff
@@ -64,6 +64,10 @@ func ParseOpenAPIErrorCloud(err error) (*V3Error, error) {
 		if err := json.Unmarshal(openAPIError.Body(), &decodedError); err != nil {
 			return nil, err
 		}
+		if decodedError.Message == "" && decodedError.Code == 0 {
+			// Sometimes the SDK puts the error message in `error` instead of `body`
+			decodedError.Message = openAPIError.Error()
+		}
```

Comment:
Specifically, if kafkarest SDK function returns a struct, then it puts whatever the current state of the return struct is in the `body` field.

It's a weird design, but unless/until we modify the SDK we'll have to work around that to catch these errors.